benchmark/definition/benchmarking.py

```diff
@@ -0,0 +1,68 @@
+"""Definition of the Benchmark class."""
+
+import time
+from collections.abc import Callable
+from datetime import datetime
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of, is_callable
+from pandas import DataFrame
+
+from benchmark.definition.config import BenchmarkScenarioSettings
+from benchmark.result.metadata_class import ResultMetadata
+from benchmark.result.result import Result
+
+BenchmarkFunction = Callable[[BenchmarkScenarioSettings], DataFrame]
+
+
+@define
+class Benchmark:
```

Comment:
@AVHopp @Scienfitz: Not sure, but should we rename this class to `BenchmarkDefinition` to be precise? Preferences?

Reply:
No strong feelings from my end, both is fine.